PokemonCollection.java

```diff
@@ -0,0 +1,145 @@
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.*;
+
+public class PokemonCollection {
+		
+	/** The current number of POKEMONs in the array */
+	private int numPOKEs;
+		
+	/** The array to contain the POKEMONs */
+	private Pokemon[] pokeArray;
+		
+	/** The name of the data file that contains poke data */
+	private String sourceName;
+		
+	/** Boolean flag to indicate whether the Pokemon collection was
+		    modified since it was last saved. */
+	private boolean modified;
+		
+	/** PokemonCollection Constructor **/
+	public PokemonCollection() {
+		numPOKEs = 0;
+		pokeArray = new Pokemon[7];
```

Comment:
Don't think we are gonna add new pokemon during runtime so 7 and doubling when array runs out of space seems out of place. Instead of 7 use numPokes and add a flag for the number of pokemon in pokemondata.txt

Reply:
this same logic could be used for moves also